webrtc/sctp/sctp_connection.nim

```diff
@@ -0,0 +1,259 @@
+# Nim-WebRTC
+# Copyright (c) 2024 Status Research & Development GmbH
+# Licensed under either of
+#  * Apache License, version 2.0, ([LICENSE-APACHE](LICENSE-APACHE))
+#  * MIT license ([LICENSE-MIT](LICENSE-MIT))
+# at your option.
+# This file may not be copied, modified, or distributed except according to
+# those terms.
+
+import nativesockets, bitops, sequtils
+import usrsctp, chronos, chronicles, stew/[ptrops, endians2, byteutils]
+import ./sctp_utils, ../errors, ../dtls/dtls_connection
+
+logScope:
+  topics = "webrtc sctp_connection"
+
+const SctpConnTracker* = "webrtc.sctp.conn"
+
+type
+  SctpConnOnClose* = proc() {.raises: [], gcsafe.}
+
+  SctpState* = enum
+    SctpConnecting
+    SctpConnected
+    SctpClosed
+
+  SctpMessageParameters* = object
```

Comment:
Can we add some docs about those types and fields here?

Reply:
Done. Let me know if it's too verbose or not.